Sources/PackageDescription/Target.swift

```diff
@@ -1032,6 +1033,40 @@ extension Target.Dependency {
     }
 }
 
+/// A condition that limits the application of a target's dependency.
+public struct TargetDependencyCondition: Encodable {
+    private let platforms: [Platform]?
+
+    private init(platforms: [Platform]?) {
+        self.platforms = platforms
+    }
+
+    /// Creates a target dependency condition.
+    ///
+    /// - Parameters:
+    ///   - platforms: The applicable platforms for this target dependency condition.
+    @_disfavoredOverload
+    @available(_PackageDescription, obsoleted: 5.7, message: "using .when with nil platforms is obsolete")
```

Comment:
Will this warning also trigger if my manifest assigns the platform to a variable of type `[Platform]?`?  I'm not sure how interesting that case is, but I could see a manifest doing that if it somehow conditionalized whether there are conditions.

Reply:
yes it will emit the availability warning and the users will need to change the code appropriately as they transition to 5.7